### PR TITLE
:pushpin: Add Netbox v4.1.* compatibility

### PR DESCRIPTION
**Pull request by fanshan**
Hello,

Thank you for developing this plugin. We would like to tests it, but we use Netbox v4.1.* versions.

For that, I propose a little patch I think is not complete. I have noted these last points to see with you in order to finalize this PR:

- [x] Review the NetBox changelog for any relevant code changes
- [x] Run automated tests and resolve any errors
- [x] Perform manual testing of UI elements
- [x] Perform manual testing of NetBox branching

If you see any other point to raise, I can update this PR as well.

### CODE REVIEW

**Comment by pheus** [comment]
Hello @fanshan,

Thank you for your interest in this plugin and for opening the PR. You're correct that this patch removes the safeguards preventing the plugin from running against NetBox 4.1, but it will require both automated and manual testing before I can accept these changes.

For this purpose, I recommend reordering and expanding the task list as follows:

- [ ] Review the NetBox changelog for any relevant code changes
- [ ] Run automated tests and resolve any errors
- [ ] Perform manual testing of UI elements
- [ ] Perform manual testing of NetBox branching
- [ ] Update plugin dependencies
- [ ] Update NetBox version requirements
- [ ] Update documentation
- [ ] Release a new version

Would you be interested in helping with testing and fixing any upgrade-related errors/bugs? In my initial testing, I encountered several GraphQL-related errors.

Otherwise, I can aim to provide a compatible version within the next few days.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
     strategy:
       matrix:
         python-version: ["3.10", "3.11", "3.12"]
-        netbox-version: ["v4.0.0", "master"]
+        netbox-version: ["v4.1.1", "master"]
```
Comment (pheus) [inline .github/workflows/ci.yaml:35]: Could you please set the version to v4.1.0 for testing against the first release of NetBox 4.1?

Comment (fanshan) [reply .github/workflows/ci.yaml:35]: Yes, I will do it.

**Comment by pheus** [comment]
Thank you very much for your help!

> Another question: do you want to keep compatibility with Netbox version inferior to v4.1.0?

At this stage of development, it is fine to raise the minimum required version to 4.1.0.

**Comment by fanshan** [comment]
Hello @pheus,

I still have this kind of error when I run the tests:

```
======================================================================
FAIL: test_graphql_get_object (netbox_aci_plugin.tests.test_api.ACIVRFAPIViewTestCase.test_graphql_get_object)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Volumes/workspace/projects/saashup/netbox/v4/venv/lib/python3.12/site-packages/django/test/utils.py", line 443, in inner
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Volumes/workspace/projects/saashup/netbox/v4/netbox/netbox/utilities/testing/api.py", line 555, in test_graphql_get_object
    self.assertIn('errors', data)
AssertionError: 'errors' not found in {'data': {'aci_vrf': {'changelog': [], 'display': 'ACIVRFTestAPI1', 'class_type': 'ACIVRF', 'tags': [], 'journal_entries': [], 'custom_fields': {}, 'aci_tenant': {'id': '43'}, 'nb_tenant': {'id': '29'}, 'nb_vrf': {'id': '12'}, 'dns_labels': ['DNS1', 'DNS2'], 'id': '28', 'created': '2024-10-01T11:59:27.402923+00:00', 'last_updated': '2024-10-01T11:59:27.402926+00:00', 'custom_field_data': {}, 'name': 'ACIVRFTestAPI1', 'name_alias': 'Testing', 'description': 'First ACI Test', 'bd_enforcement_enabled': False, 'ip_data_plane_learning_enabled': True, 'pc_enforcement_direction': 'ingress', 'pc_enforcement_preference': 'unenforced', 'pim_ipv4_enabled': False, 'pim_ipv6_enabled': False, 'preferred_group_enabled': False, 'comments': '# ACI Test 1'}}}
```

Does this error tell you something?

**Comment by pheus** [comment]
Hi @fanshan,

I was able to narrow down the problem. It’s actually related to a GraphQL permission change introduced in v4.0.11.

Here is the relevant PR:  
https://github.com/netbox-community/netbox/pull/17312

I will update the code to ensure compatibility with v4.0.11.

**Comment by pheus** [comment]
Hi @fanshan,

I was able to narrow down the problem. It's actually related to GraphQL changes introduced in v4.0.10 and v4.0.11.

Here are the relevant PRs:
https://github.com/netbox-community/netbox/pull/17244
https://github.com/netbox-community/netbox/pull/17312

I will update the code to ensure compatibility with v4.0.11.

**Comment by pheus** [comment]
Hi @fanshan,

It seems that with these changes, there are no more errors with NetBox versions 4.1.0, 4.1.1, and 4.1.2.
I hope you didn’t invest too much time troubleshooting the tests. I actually wanted to support you in finding the root cause, but I had to fix the tests myself to reproduce the error.

If you're still interested, you can update your fork and rebase your changes on top. Afterward, I would be happy to accept your PR.

Nonetheless, I really want to thank you for your time and effort.

**Comment by fanshan** [comment]
Hello @pheus,

It's was fine ! Tests pass and the plugin is improved :-)

Branch rebased. Let me know if you want to change the plugin version.

**Comment by pheus** [comment]
Hello @fanshan,

Thank you very much for this great work! 🚀 
I will merge your PR and take it from here.

**Review by pheus — APPROVED** [review]
Well done! Thank you very much 🥇

**Pull request closed without merging**